From bfceb97239e8769bba5df637913da4f59f557124 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Mon, 5 Jun 2017 19:47:17 +0800
Subject: [PATCH 1852/5242] MLK-15009 irqchip: imx-irqsteer: correct type of
 irqstat

commit  fcf530c98cd349e246226e4bcbc2b116cf7449ec from
https://source.codeaurora.org/external/imx/linux-imx.git

The type of irqstat in irqsteer_irqchip_data unsigned long, actually
it needs to be 32bits width, so use unsigned int.

And use sizeof(irqsteer_data->irqstat[0]), instead of 4 when alloc
memory for irqsteer_data.

for_each_set_bit needs the second param type is unsigned long *, so
cast the irqsteer_data->irqstat to unsigned long *, this is safe here.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/irqchip/irq-imx-irqsteer.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/irqchip/irq-imx-irqsteer.c b/drivers/irqchip/irq-imx-irqsteer.c
index 1a961f4..bc16453 100644
--- a/drivers/irqchip/irq-imx-irqsteer.c
+++ b/drivers/irqchip/irq-imx-irqsteer.c
@@ -32,7 +32,7 @@ struct irqsteer_irqchip_data {
 	int irq;
 	int channum;
 	struct irq_domain *domain;
-	unsigned long irqstat[];
+	unsigned int irqstat[];
 };
 
 static void imx_irqsteer_irq_unmask(struct irq_data *d)
@@ -130,7 +130,7 @@ static void imx_irqsteer_irq_handler(struct irq_desc *desc)
 	imx_irqsteer_update_irqstat(irqsteer_data);
 
 	irqnum = irqsteer_data->channum * 32;
-	for_each_set_bit(pos, irqsteer_data->irqstat, irqnum) {
+	for_each_set_bit(pos, (unsigned long *)irqsteer_data->irqstat, irqnum) {
 		virq = irq_find_mapping(irqsteer_data->domain, pos);
 		if (virq)
 			generic_handle_irq(virq);
@@ -153,7 +153,8 @@ static int imx_irqsteer_probe(struct platform_device *pdev)
 		channum = 1;
 
 	irqsteer_data = devm_kzalloc(&pdev->dev, sizeof(*irqsteer_data) +
-				     channum * 4,
+				     channum *
+				     sizeof(irqsteer_data->irqstat[0]),
 				     GFP_KERNEL);
 	if (!irqsteer_data)
 		return -ENOMEM;
-- 
1.7.9.5

